### PR TITLE
Adding support to add and remove multiple slides

### PR DESCRIPTION
**Pull request by seriousben**
Adding simple support to and remove multiple slides.

Just pass an array of elements, html strings to slickAdd.
Just pass an array of indexes to slickRemove.

Basically getting rid of reinit every add/remove.

### CODE REVIEW

**Comment by kenwheeler** [comment]
Can you update README? Have you tested this everywhere?

**Comment by simeydotme** [comment]
@seriousben this looks like a good contribution, but need some serious testing.
If you're unable to provide fiddles and decent test coverage in them, I'll close this in the coming week. :)

Hope you got some time, else I'll see if we can add it to 2.0

**Comment by simoami** [comment]
This is a great and important feature. We're incorporating remote pagination, and the reinit in each slide is a real performance killer!

screen shot 2016-04-13 at 10 56 17 am

**Comment by kenwheeler** [comment]
@simoami have you tried this branch personally?

File: slick/slick.js [hunk 1]
```diff
@@ -186,10 +186,21 @@
...
-    Slick.prototype.addSlide = function(markup, index, addBefore) {
+    Slick.prototype.addSlide = function(markup, index, addBefore, skipReinit) {
+
 
         var _ = this;
 
+
+        if ($.isArray(markup)) {
+            $.each(markup, function(idx, markupItem) {
+                _.addSlide(markupItem, index, addBefore, true);
```
Comment (simoami) [inline slick/slick.js]: if index was specified, wouldn't you want to increment the index for each iteration in the array?

```
_.addSlide(markupItem, index++, addBefore, true);
```

**Comment by simoami** [comment]
Added a code comment:
https://github.com/kenwheeler/slick/pull/876/commits/422065e95f90a7c842f7bcf68cab5db195801385#r59568539

**Comment by simoami** [comment]
@kenwheeler Not yet. I'm about to.

**Comment by seriousben** [comment]
Unfortunately I won't have time to support this PR. Please take it over if it works for you.

File: slick/slick.js [hunk 2]
```diff
@@ -1236,10 +1248,19 @@
 
     };
 
-    Slick.prototype.removeSlide = function(index, removeBefore, removeAll) {
+    Slick.prototype.removeSlide = function(index, removeBefore, removeAll, skipRenit) {
```
Comment (simoami) [inline slick/slick.js]: proper spelling is "skipReinit"

**Comment by simoami** [comment]
Thanks, I'll spawn a new PR

**Comment by simoami** [comment]
I know this is closed. but just want to update that the PR as is needs some more work. For example, you can't skip reinit() without also skipping unload(). They need to be executed together or not, otherwise you might end up with duplicate arrows. unload() removes existing arrows and reinit reinserts them.

**Pull request closed without merging**